trie/trieStorageManager.go

```diff
@@ -184,17 +197,28 @@ func (tsm *trieStorageManager) Get(key []byte) ([]byte, error) {
 		return nil, core.ErrContextClosing
 	}
 
-	val, err := tsm.mainStorer.Get(key)
+	val, foundInCache, err := tsm.mainStorer.GetWithStats(key)
```

Comment:
I don't think the trieStorageManager should know about the caches... could the statsCollector be injected one level lower in a PruningStorerWithStats for example?